libs/agno/agno/agent/agent.py

```
@@ -1407,18 +1407,13 @@ def add_tools_to_model(self, model: Model) -> None:
                 model.set_functions(functions=self._functions_for_model)
 
     def update_model(self) -> None:
-        # Use the default Model (OpenAIChat) if no model is provided
+        #raise the error to provide the model instead of defaulting any model
         if self.model is None:
-            try:
-                from agno.models.openai import OpenAIChat
-            except ModuleNotFoundError as e:
-                logger.exception(e)
-                logger.error(
-                    "Agno agents use `openai` as the default model provider. "
-                    "Please provide a `model` or install `openai`."
-                )
-                exit(1)
-            self.model = OpenAIChat(id="gpt-4o")
+            logger.error(
```

You could probably make the `model` parameter of the class as required instead of Optional.
Also this will break tons of code which do not provide model when creating Agent (like most of the examples in cookbook/tools section)
I think we should create a discussion in the Issues to talk about it. Also, if it is merged, it should be released in the next major patch as it is a breaking change.

---

True that, we should not have default pointing to a single vendor.
We can probably have a default config where we can define the default model behaviour. Like whether to raise an error in case the model is not provided or use a default model (which can be configured there). That will give lots of flexibility to the users of Agno. Something like a .agno_config file or similar.
Again, would be good to discuss this further.

---

I think if we do that, there’s no point in having these kinds of updates if you still need to configure it. We can include the model in every agent, and we won’t face these issues. My point is that we shouldn’t default to any single vendor. Instead, we can raise an error if the model isn’t provided. After making this change, we’ll only need to make small adjustments across the codebase, and it won’t take much time.